content/docs/scaffolder/manipulate-yaml-file/index.md

```diff
@@ -0,0 +1,177 @@
+spec:
+  owner: group:default/engineering
+  type: service
+
+  steps:
+    - id: log-message
+      name: Log Message
+      action: debug:log
+      input:
+        message: Template execution started
+
+    - id: create-file
+      name: Create file
+      action: roadiehq:utils:fs:write
+      input:
+        path: ./template.yaml
+        content: >
+          extend:
+            me: i am a random yaml
+            this: will be extended by the scaffolder action
+
+    - id: merge
+      name: Merge
+      action: roadiehq:utils:merge
+      input:
+        path: ./template.yaml
+        content: >
+          foo: bar
+
+    - id: parse
+      name: read file
+      action: roadiehq:utils:fs:parse
+      input:
+        path: ./template.yaml
+        parser: yaml
+
+    - id: serialize
+      name: Serialize
+      action: roadiehq:utils:serialize:yaml
+      input:
+        data: ${{ steps['parse'].output.content }}
+
+    - id: log
+      name: Console.log
+      action: debug:log
+      input:
+        message: |
+          Content: `${{ steps['serialize'].output.serialized }}`
+```
+
+## Breakdown
+
+### Parameters
+
+We didnt use any parameters for this demonstration as we don't need any user input to showcase the power of these actions.
+
+### Steps
+
+#### debug:log
+
+Uses the debug:log action to output a log message to the backstage scaffolder UI. The message here is a general hard-coded message that indicates that the template execution has started
+
+```yaml
+- id: log-message
+  name: Log Message
+  action: debug:log
+  input:
+    message: Template execution started
+```
+
+#### roadiehq:utils:fs:write
+
+```yaml
+- id: create-file
+  name: Create file
+  action: roadiehq:utils:fs:write
+  input:
+    path: ./template.yaml
+    content: >
+      extend:
+        me: i am a yaml
+        this: will be extended by the scaffolder action
+```
+
+We use this action to create a basic yaml file into the root of the workspace with the content:
+
+```yaml
+# template.yaml
+extend:
+  me: i am a yaml
+  this: will be extended by the scaffolder action
+```
+
+#### roadiehq:utils:merge
+
+```yaml
+- id: merge
+  name: Merge
+  action: roadiehq:utils:merge
+  input:
+    path: ./template.yaml
+    content: >
+      foo: bar
+```
+
+This action will merge the `input.content` into the file you specify in the `input.path` property. After the merge our template.yaml file will look like this:
+
+```yaml
+# template.yaml
+extend:
+  me: i am a yaml
+  this: will be extended by the scaffolder action
+foo: bar
+```
+
+#### roadiehq:utils:fs:parse
+
+```yaml
+- id: parse
+  name: read file
+  action: roadiehq:utils:fs:parse
+  input:
+    path: ./template.yaml
+    parser: yaml
+```
+
+We use this action to parse the file specified at `input.path` because we know it is a yaml file we can use the `input.parser` with the value `yaml`. This will make the content of the file to be parsed into an object properly
```

Comment:
```suggestion
We use this action to parse the file specified at `input.path` because we know it is a YAML file we can use the `input.parser` with the value `yaml`. This will make the content of the file to be parsed into an object properly.
```